Backpack/src/androidTest/java/net/skyscanner/backpack/button/BpkButtonTest.kt

```diff
@@ -24,23 +35,39 @@ class BpkButtonTest {
   // The drawables are set as start,top,end,bottom and are accessible in the compoundDrawables array
   @Test
   fun test_icon_end() {
-    val context = InstrumentationRegistry.getInstrumentation().targetContext
     val trainIcon = AppCompatResources.getDrawable(context, R.drawable.bpk_train)
     val button = BpkButton(context).apply {
       icon = trainIcon
       iconPosition = BpkButton.END
     }
-    Assert.assertEquals(button.compoundDrawables[2],trainIcon)
+    Assert.assertEquals(button.compoundDrawables[2], trainIcon)
   }
 
   @Test
   fun test_icon_start() {
-    val context = InstrumentationRegistry.getInstrumentation().targetContext
     val trainIcon = AppCompatResources.getDrawable(context, R.drawable.bpk_train)
     val button = BpkButton(context).apply {
       icon = trainIcon
       iconPosition = BpkButton.START
     }
-    Assert.assertEquals(button.compoundDrawables[0],trainIcon)
+    Assert.assertEquals(button.compoundDrawables[0], trainIcon)
+  }
+
+  @Test
+  fun givenBpkButtonInAnyTypeAndFormWhenIsEnabledIsSetThenSetupIsCalled() {
```

Comment:
can we change this to normal unit testing? We're not using BDD here.

Reply:
context is needed if you want to inflate the view. Tried mocking it but Android framework doesn't allow for that.

Reply:
I follow the convention we use generally in Skyscanner for Android. I don't see any real value in the naming convention used above in the other tests but feel free to change my test name to whatever it would suit this project.

Reply:
You may rename it to 'test_enabled_state'. Since this PR is from a fork, I won't be able to change it.